### PR TITLE
Set allowed authenticators in tools and via env variables

### PR DESCRIPTION
**Pull request by gow**
## What changed?
https://github.com/temporalio/temporal/pull/6155 introduced the config value `allowedAuthenticators`. This PR updates the `temporal-cassandra-tool` to use this config value. It also updates `config_template.yaml` so that the config value can be set in docker images as well.

## Why?
Cassandra supports several authenticators to verify the login credentials. Gocql implements a subset of them as [default allowed list of authenticators](https://github.com/gocql/gocql/blob/34fdeebefcbf183ed7f916f931aa0586fdaa1b40/conn.go#L27). It allows customization of this list to allow supporting additional authenticators.
So with this change Temporal can connect to Cassandra clusters that have authenticators other than the default ones provided by Gocql.

## How did you test it?
Ran `./temporal-cassandra-tool` and verified that it picked up the values in environment variable `CASSANDRA_ALLOWED_AUTHENTICATORS`. Couldn't test docker build since it's in a different repo.

## Is hotfix candidate?
No

### CODE REVIEW

File: docker/config_template.yaml [hunk 1]
```diff
@@ -20,6 +20,7 @@ persistence:
                 keyspace: "{{ default .Env.KEYSPACE "temporal" }}"
                 user: "{{ default .Env.CASSANDRA_USER "" }}"
                 password: "{{ default .Env.CASSANDRA_PASSWORD "" }}"
+                allowedAuthenticators: "{{ default .Env.CASSANDRA_ALLOWED_AUTHENTICATORS "" }}"
```
Comment (robholland) [inline docker/config_template.yaml:23]: This won't unmarshal properly, the yaml needs to provide a list here. I tested by adding this to `config/development.yaml`:

```
  datastores:
    cassandra:
      cassandra:
        allowedAuthenticators: "one, two"
```

and then running the `config/config_test.go`

```--- FAIL: TestToString (0.00s)
    /Users/robholland/Developer/github.com/temporalio/temporal/common/config/config_test.go:37: 
        	Error Trace:	/Users/robholland/Developer/github.com/temporalio/temporal/common/config/config_test.go:37
        	Error:      	Received unexpected error:
        	            	yaml: unmarshal errors:
        	            	  line 12: cannot unmarshal !!str `one, two` into []string
        	Test:       	TestToString
```

You will need to use use a range and dockerize's `split` helper here to provide an array with a value per authenticator.

File: tools/cassandra/main.go [hunk 2]
```diff
@@ -86,6 +86,12 @@ func buildCLIOptions() *cli.App {
 			Usage:  "Password used for authentication for connecting to cassandra host",
 			EnvVar: "CASSANDRA_PASSWORD",
 		},
+		cli.StringSliceFlag{
+			Name:   schema.CLIFlagAllowedAuthenticators,
+			Value:  nil,
+			Usage:  "List of authenticators allowed to be used by the gocql client while connecting to the server.",
```
[the remark this answers is not in the document]
Comment (gow) [reply tools/cassandra/main.go:92]: This list is for the Gocql client only (not cassandra server). Here is the default list they support - https://github.com/gocql/gocql/blob/34fdeebefcbf183ed7f916f931aa0586fdaa1b40/conn.go#L27. 

This config overrides that list. Unfortunately there is no option to merging the allowed list of authenticators.

[the remark this answers is not in the document]
Comment (gow) [reply tools/cassandra/main.go:92]: Since it's not exposed either, we can't do the merging of the lists.

File: docker/config_template.yaml (hunk 1 above)
Comment (gow) [reply docker/config_template.yaml]: Good catch. Fixed it. Here is how I tested it.
1. `export CASSANDRA_ALLOWED_AUTHENTICATORS="one, two"`
2. `dockerize -template ./docker/config_template.yaml:/tmp/qqq.yaml && cat /tmp/qqq.yaml` - Verify that the allowedAuthenticators field is populated as a list.
3. Update `config/development.yaml` with the values generated above and run - `go test ./common/config`

Comment (gow) [reply docker/config_template.yaml]: I think we should have config tests for cassandra as well in our unit tests. It's outside the scope of this PR and needs a bit of test setup. I'll add it separately.

Comment (robholland) [reply docker/config_template.yaml]: I agree about the need and being out of scope.

File: docker/config_template.yaml [hunk 3]
```diff
@@ -20,6 +20,11 @@ persistence:
                 keyspace: "{{ default .Env.KEYSPACE "temporal" }}"
                 user: "{{ default .Env.CASSANDRA_USER "" }}"
                 password: "{{ default .Env.CASSANDRA_PASSWORD "" }}"
+                {{- if .Env.CASSANDRA_ALLOWED_AUTHENTICATORS }}
+                allowedAuthenticators: {{ range split .Env.CASSANDRA_ALLOWED_AUTHENTICATORS "," }}
+                    - {{.}}
```
Comment (robholland) [inline docker/config_template.yaml:25]:
```suggestion
                    - {{trim .}}
```

Comment (robholland) [reply docker/config_template.yaml:25]: It's not documented for some reason, but dockerize supports all of the Sprig functions: https://github.com/Masterminds/sprig/blob/581758eb7d96ae4d113649668fa96acc74d46e7f/functions.go#L97

**Review by robholland — APPROVED** [review]
Looks good, just one suggestion to get rid of the whitespace issue.

**Review by bergundy — APPROVED** [review]

**Pull request merged**